lib/sparkle_formation/utils.rb

```diff
@@ -3,6 +3,43 @@ class SparkleFormation
   # Helper utilities
   module Utils
 
+    module TypeCheckers
+
+      # Validate given value is type defined within valid types
+      #
+      # @param val [Object] value
+      # @param types [Class, Array<Class>] valid types
+      # @return [NilClass]
+      # @raises [TypeError]
+      def __t_check(val, types)
+        types = [types] unless types.is_a?(Array)
+        if(types.none?{|t| val.is_a?(t)})
+          file_name, line_no = ::Kernel.caller.detect do |l|
+            !l.split(':').first.to_s.include?('/sparkle_formation')
```

Comment:
Because if someone is using a `sparkle_formation` sub-directory or file name it will result in a false positive.